src/Dialect/ONNX/ElementsAttr/ElementsAttrBuilder.cpp

```diff
@@ -497,7 +500,231 @@ bool isIdentityPermutation(ArrayRef<uint64_t> perm) {
+        auto dstSrcPairsIter = dstSrcPositionPairs.begin();
+        for (auto &idxoffs : StridesRange<1>(inputShape, {inputStrides})) {
+          auto idx = idxoffs.index;
+          if ((batchIndex == 1) &&
+              ((idx[1] == seqLengthIndex) &&
+                  (idx[0] < sequenceLength[seqLengthIndex]))) {
+            auto listIter2 = reversePosList2.begin();
+            std::advance(listIter2, idx[0]);
+            (*listIter2).push_back(idxoffs[0]);
+            // This will be same as the correspoding pos of list in
+            // reversePosList1
+            int posIndex = (*listIter2).size() - 1;
+            // get the iterator for same position in the reversePosList1
+            auto iter1 = reversePosList1.begin();
+            // Here we are getting the list from end.
+            // analogy slice(0, 3, None) <-> slice(2, None, -1)
+            // for lists 0,1,2 we will get 2,1,0
+            std::advance(iter1, (reversePosList1.size() - 1 - idx[0]));
+            auto innerListIter = (*iter1).begin();
+            std::advance(innerListIter, posIndex);
+            // this is the source pos that needs to be assigned.
+            (*dstSrcPairsIter).second = *(innerListIter);
+            dstSrcPairsIter++;
+          } else if ((batchIndex == 0) &&
+                     ((idx[0] == seqLengthIndex) &&
+                         (idx[1] < sequenceLength[seqLengthIndex]))) {
+            auto iter2 = reversePosList2.begin();
+            std::advance(iter2, idx[1]);
+            (*iter2).push_back(idxoffs[0]);
+            int posIndex = (*iter2).size() - 1;
+            auto iter1 = reversePosList1.begin();
+            std::advance(iter1, (reversePosList1.size() - 1 - idx[1]));
+            auto innerListIter = (*iter1).begin();
+            std::advance(innerListIter, posIndex);
+            (*dstSrcPairsIter).second = *(innerListIter);
+            dstSrcPairsIter++;
+          }
+        }
+        if (enableVerbose) {
+          std::cout << " Printing the assigned dstSrcPositionPairs "
+                    << std::endl;
+          int ipos = 0;
+          for (auto it : dstSrcPositionPairs) {
+            std::cout << ipos++ << " : " << it.first << "  -- " << it.second
+                      << std::endl;
+          }
+        }
+        auto dstSrcLookupIter = dstSrcPositionPairs.begin();
+        // Populating the destination with correct values.
+        for (auto &idxoffs : StridesRange<1>(inputShape, {inputStrides})) {
+
+          int64_t pos = idxoffs[0];
+          // std::cout << " pos " << pos << " idxoffs.flattenedIndex "
+          //           << idxoffs.flattenedIndex << " : "
+          //           << (*dstSrcLookupIter).first << " - "
+          //           << (*dstSrcLookupIter).second << std::endl;
+
+          if (pos < ((*dstSrcLookupIter).first)) {
+            continue;
+          }
+          if (pos == (*dstSrcLookupIter).first) {
+            cpptype accumulator = 0;
```

Comment:
```suggestion
            cpptype accumulator = 0;
```
Naming: This does not accumulate

Reply:
True. renamed it now.